### PR TITLE
Full null safety

### PR DESCRIPTION
**Pull request by adamwalzer**
Support full null safety and convert functions to support both sync and async.

### CODE REVIEW

File: lib/lzstring.dart [hunk 1]
```diff
@@ -50,200 +61,162 @@ class LZString {
...
-
-      for (ii = 0; ii < uncompressed.length; ii++) {
-        contextC = uncompressed[ii];
-        if (!contextDictionary.containsKey(contextC)) {
-          contextDictionary[contextC] = contextDictSize++;
-          contextDictionaryToCreate[contextC] = true;
-        }
+  /// Synchronously produces invalid UTF-16 strings from [uncompressed].
+  ///
+  /// Can be decompressed with `decompress`.
```
Comment (skipness) [inline lib/lzstring.dart]: Could you also update all inline code block to square brackets in rest of the doc?

**Review by skipness — COMMENTED** [review]
Code LGTM

File: lib/lzstring.dart (hunk 1 above)
Comment (adamwalzer) [reply lib/lzstring.dart]: Sure thing

Comment (adamwalzer) [reply lib/lzstring.dart]: Fixed

**Review by skipness — APPROVED** [review]

**Pull request merged**